src/completions.js

```diff
@@ -33,15 +34,25 @@ export class Completion {
 }
 
 // Normal completions are returned just like spec completions
-export class NormalCompletion extends Completion {}
+export class NormalCompletion extends Completion {
+  constructor(value: Value, location: ?BabelNodeSourceLocation, target?: ?string) {
+    super(value, location, target);
+    invariant(this.constructor.name !== "NormalCompletion", "NormalCompletion is an abstract base class");
```

Comment:
You should be able to check `this.constructor !== NormalCompletion`. (And same for AbruptCompletion.)